packages/web-app-files/tests/unit/views/PublicLink.spec.js

```diff
@@ -1,34 +1,179 @@
+
+      beforeEach(async () => {
+        await wrapper.setData({ passwordRequired: true })
+        passwordRequiredForm = wrapper.find('form')
+      })
+
+      it('should display the password required form', () => {
+        expect(passwordRequiredForm.exists()).toBeTruthy()
+        expect(passwordRequiredForm).toMatchSnapshot()
+      })
+
+      describe('and the password input', () => {
+        it('should have a computed label', () => {
+          const passwordInput = passwordRequiredForm.find(textInputStub)
+
+          expect(passwordInput).toMatchSnapshot()
+        })
+        it('should not display the error message if "inputErrorMessage" is falsy', () => {
+          const passwordInput = passwordRequiredForm.find(textInputStub)
+
+          expect(passwordInput.attributes().errormessage).toBeUndefined()
+        })
+        it('should display the error message if "inputErrorMessage" is not empty', async () => {
+          await wrapper.setData({ inputErrorMessage: 'some input error message' })
+          const passwordInput = passwordRequiredForm.find(textInputStub)
+
+          expect(passwordInput).toMatchSnapshot()
+        })
+      })
+
+      describe('and the submit button', () => {
+        it('should be set as disabled if "password" is empty', () => {
+          expect(wrapper.vm.password).toBeNull()
 
-    describe('when "passwordRequired" is set as true', () => {
-      test.todo('should display the password required form')
+          const submitButton = wrapper.find(selectors.submitButton)
+          expect(submitButton.exists()).toBeTruthy()
+          expect(submitButton).toMatchSnapshot()
+        })
+        it('should be set as enabled if "password" is not empty', async () => {
+          await wrapper.setData({ password: 'some-value' })
+          expect(wrapper.vm.password).toBe('some-value')
 
-      describe('password input', () => {
-        test.todo('should have a computed label')
-        test.todo('should display the error message if "inputErrorMessage" is not empty')
-        test.todo('should not display the error message if "inputErrorMessage" is falsy')
+          const submitButton = wrapper.find(selectors.submitButton)
+          expect(submitButton.exists()).toBeTruthy()
+          expect(submitButton.attributes().disabled).toBeUndefined()
+        })
       })
 
-      describe('submit button', () => {
-        test.todo('should be set as disabled if "password" is empty')
-        test.todo('should be set as enabled if "password" is not empty')
+      it('should call "resolvePublicLink" method on form submit', async () => {
+        const spyResolvePublicLink = jest.spyOn(PublicLink.methods, 'resolvePublicLink')
+        const wrapper = getMountedWrapper({ loading: false })
+
+        await wrapper.setData({ passwordRequired: true, password: 'some-pass' })
+
+        const submitButton = wrapper.find(selectors.submitButton)
+        await submitButton.trigger('submit')
```

Comment:
@kiranparajuli589 I was able to remove your [workaround](https://stackoverflow.com/questions/53382235/trigger-form-submit-on-button-click-in-vue-unit-test) by switching from `.trigger('click')` which is correct from a users perspective to a `.trigger('submit')` which is the technically correct case since we're submitting a form 🤷🏽‍♂️ that was unexpected!